modules/core/src/v2/coins/avaxcToken.ts

```diff
@@ -0,0 +1,83 @@
+/**
+ * @prettier
+ */
+import { BitGo } from '../../bitgo';
+
+import { AvaxC } from './avaxc';
+import { CoinConstructor } from '../coinFactory';
+import { coins } from '@bitgo/statics';
+
+export interface AvaxcTokenConfig {
+  name: string;
+  type: string;
+  coin: string;
+  network: string;
+  tokenContractAddress: string;
+  decimalPlaces: number;
+}
+
+export class AvaxCToken extends AvaxC {
+  public readonly tokenConfig: AvaxcTokenConfig;
+
+  constructor(bitgo: BitGo, tokenConfig: AvaxcTokenConfig) {
+    const staticsCoin = tokenConfig.network === 'Mainnet' ? coins.get('avaxc') : coins.get('tavaxc');
+    super(bitgo, staticsCoin);
+    this.tokenConfig = tokenConfig;
+  }
+
+  static createTokenConstructor(config: AvaxcTokenConfig): CoinConstructor {
+    return (bitgo: BitGo) => new AvaxCToken(bitgo, config);
+  }
+
+  get type(): string {
+    return this.tokenConfig.type;
+  }
+
+  get name(): string {
+    return this.tokenConfig.name;
+  }
+
+  get coin(): string {
+    return this.tokenConfig.coin;
+  }
+
+  get network(): string {
+    return this.tokenConfig.network;
+  }
+
+  get tokenContractAddress(): string {
+    return this.tokenConfig.tokenContractAddress;
+  }
+
+  get decimalPlaces(): number {
+    return this.tokenConfig.decimalPlaces;
+  }
+
+  getChain(): string {
```

Comment:
the same as `type()`, why do we need two?

Reply:
This is how it is designed.

https://github.com/BitGo/BitGoJS/blob/STLX-10425-add-avaxcToken-in-SDK/modules/core/src/v2/baseCoin.ts#L258-L265